scripts/utils.ts

```diff
@@ -1,5 +1,61 @@
-import type { EventLog, Log } from "ethers";
+import { ethers } from "hardhat";
+import { EventLog, Log, FeeData } from "ethers";
 
+/**
+ * Response for current gas fee data from the Amoy gas station API at:
+ * https://gasstation-testnet.polygon.technology/amoy
+ */
+interface AmoyFeeData {
+  safeLow: {
+    maxPriorityFee: number;
+    maxFee: number;
+  };
+  standard: {
+    maxPriorityFee: number;
+    maxFee: number;
+  };
+  fast: {
+    maxPriorityFee: number;
+    maxFee: number;
+  };
+  estimatedBaseFee: number;
+  blockTime: number;
+  blockNumber: number;
+}
+
+/**
+ * Fetches the current gas fee data from the Amoy Gas Station API.
+ * @param url Amoy gas station API URL.
+ * @returns Current gas fee information or `null` (to align with ethers
+ * `getFeeData` method return type)
+ */
+export async function getFeeData(chainName: string): Promise<FeeData> {
+  if (chainName === "polygon-amoy") {
+    try {
+      const url = "https://gasstation-testnet.polygon.technology/amoy";
+      const response = await fetch(url);
+      const data = (await response.json()) as AmoyFeeData;
+      const feeData = new FeeData(
+        null, // No gas price value needed
+        BigInt(Math.ceil(data.fast.maxFee * 1e9)),
```

Comment:
i _think_ this is the correct conversion from gwei to wei. e.g., the gas station will give you a value like `"maxPriorityFee": 301.778399988,
"maxFee": 301.778400003`